packages/front-end/components/Settings/EditOrganizationModal.tsx

```diff
@@ -1,41 +1,65 @@
 import { FC } from "react";
 import { useForm } from "react-hook-form";
 import { useAuth } from "@/services/auth";
+import usePermissions from "@/hooks/usePermissions";
 import Field from "../Forms/Field";
 import Modal from "../Modal";
 
 const EditOrganizationModal: FC<{
   name: string;
+  ownerEmail: string;
   close: () => void;
   mutate: () => Promise<unknown>;
-}> = ({ close, mutate, name }) => {
+}> = ({ close, mutate, name, ownerEmail }) => {
   const { apiCall, setOrgName } = useAuth();
 
+  const permissions = usePermissions();
+  const canEditOwner = permissions.check("organizationSettings");
+
   const form = useForm({
     defaultValues: {
       name,
+      ownerEmail,
     },
   });
 
   return (
     <Modal
-      header="Edit Organization Name"
+      header="Edit Organization"
       open={true}
       close={close}
       submit={form.handleSubmit(async (value) => {
+        if (
+          !value?.name ||
+          value?.name.trim() === "" ||
+          value?.name === undefined
+        ) {
+          throw new Error("Organization name cannot be empty");
+        }
+        if (!value?.ownerEmail || value.ownerEmail.trim() === "") {
+          throw new Error("Owner email cannot be empty");
+        }
         await apiCall("/organization", {
           method: "PUT",
           body: JSON.stringify(value),
         });
         // Update org name in global context (e.g. top nav)
-        // @ts-expect-error TS(2722) If you come across this, please fix it!: Cannot invoke an object which is possibly 'undefin... Remove this comment to see the full error message
-        setOrgName(value.name);
+        if (setOrgName) {
+          setOrgName(value.name);
+        }
         // Update org name on settings page
         await mutate();
       })}
       cta="Save"
     >
       <Field label="Organization Name" required {...form.register("name")} />
+      <Field
+        label={"Owner Email"}
+        type="email"
+        {...form.register("ownerEmail")}
+        disabled={!canEditOwner}
+        title={canEditOwner ? "" : "Only admins can change this"}
```

Comment:
May not need this because users without `organizationSettings` permission can't access the page, but probably better to be safe

Reply:
True, I originally had this split out into a different permission, but decided to combine it with organizationSettings. In case we want to split this again, there is a state for 'can edit name but not owner'